### ISSUE TITLE
Supporting run with a closure

### ISSUE THREAD

**Issue by greyblake** [matklad/arbtest#1]
Hi, thanks for the nice library!
I've started using it recently and now expanding it to my work related and private projects.

## What's desired

Would it be possible to make `builder.run()` accept a real closure instead of 
```rust
pub type Property = fn(u: &mut Unstructured<'_>) -> arbitrary::Result<()>;
```
In other words, something like:
```rust
pub trait Property: FnMut(&mut Unstructured<'_>) -> arbitrary::Result<()> {
}

pub fn run<P: Property>(mut self, prop: P) {
    // implementation..
}
```
## Why it would be useful

One of my tests needs to be ran with some predefined IDs, that come from another place (in particular from DB).

## Why it's not possible at the moment

I've tried to do the desired change and open a PR, but I see that it narrows down to `catch_unwind`,  which I could not workaround:

```rust
    fn fails(&mut self, seed: Seed, prop: Property) -> bool {
        let this = AssertUnwindSafe(self);
        std::panic::catch_unwind(move || {            // <- FnMut cannot be moved here
            let that = this;
            that.0.single_run(seed, prop)
        })
        .is_err()
    }
```

**Comment by matklad** [comment]
Yeah, I think `fn` pointer here is just me taking a shortcut. Some wishlist for me: 

* it should be `FnMut`, not our custom trait
* it it shouldn't force monomorphisation onto the caller (its ok to use `impl FnMut` in the API, but that should delegate to `dyn FnMut`)

**Comment by matklad** [comment]
Fixed in 0.2
